lib/actions/jlinkTargetActions.js

```diff
@@ -235,189 +246,242 @@ export function canWrite() {
-            // device and avoid resetting, that will be done by pc-nrfjprog-js
-            if ([...pages.keys()].find(addr => addr >= uicrBaseAddr) !== undefined) {
-                dispatch(writeHex(serialNumber, pages.asHexString(64), {
-                    chip_erase_mode: nrfjprog.ERASE_PAGES_INCLUDING_UICR,
-                    reset: false,
-                }));
-            } else {
-                dispatch(writeHex(serialNumber, pages.asHexString(64)));
-            }
+    return undefined;
+};
 
-            dispatch(targetActions.updateTargetWritable());
-        }).catch(() => {});
-    };
-}
+// Does some sanity checks, joins the loaded HEX files, flattens overlaps,
+// paginates the result to fit flash pages, and calls writeHex()
+export const write = () => async (dispatch, getState) => {
+    const { serialNumber, deviceInfo: { cores } } = getState().app.target;
+    const results = [];
+    const arguements = cores.map(c => ([c]));
+    return sequence(
+        args => dispatch(writeOneCore(args)),
+        results,
+        ...arguements,
+    ).then(async () => {
+        await dispatch(loadDeviceInfo(serialNumber));
+        dispatch(targetActions.writingEndAction());
+        dispatch(targetActions.updateTargetWritable());
+    });
+};
 
 // Erase all on device and write file to it.
-export function recoverAndWrite() {
-    return dispatch => dispatch(recover(true))
-        .then(() => dispatch(write()));
-}
+export const recoverAndWrite = () => dispatch => dispatch(recover(true))
+    .then(() => dispatch(write()));
 
 // Save the content from the device momery as hex file.
-export function saveAsFile() {
-    return (dispatch, getState) => {
-        const { memMap, deviceInfo } = getState().app.target;
-
-        const options = {
-            title: 'Save memory as file',
-            defaultPath: `nRF_Connect_Programmer_${Date.now()}.hex`,
-        };
-
-        const save = filePath => {
-            if (filePath) {
-                fs.writeFile(filePath, memMap.slice(0, deviceInfo.romSize).asHexString(), err => {
-                    if (err) {
-                        logger.error(`Error when saving file: ${err}`);
-                    }
-                    logger.info(`File is successfully saved at ${filePath}`);
-                });
-            }
-        };
-        remote.dialog.showSaveDialog(options, save);
+export const saveAsFile = () => (dispatch, getState) => {
+    const { memMap, deviceInfo } = getState().app.target;
+    const maxAddress = Math.max(
+        ...deviceInfo.cores.map(c => c.romBaseAddr + c.romSize),
+    );
+
+    const options = {
+        title: 'Save memory as file',
+        defaultPath: `nRF_Connect_Programmer_${Date.now()}.hex`,
+    };
+
+    const save = filePath => {
+        if (filePath) {
+            fs.writeFile(filePath, memMap.slice(0, maxAddress).asHexString(), err => {
+                if (err) {
+                    logger.error(`Error when saving file: ${err}`);
```

Comment:
When writeFiles fails I think you get an error object, you might want to check this and call logger only with err.message.